tests/recursive_call/src/main.rs

```diff
@@ -1,56 +1,36 @@
-#![no_main]
+#![cfg_attr(target_os = "zkvm", no_main)]
+
+#[cfg(target_os = "zkvm")]
 athena_vm::entrypoint!(main);
 
-use athena_vm::helpers::bytes32_to_32bit_words;
-use athena_vm::syscalls::{read_storage, write_storage};
-use athena_vm::types::{
-  StorageStatus::StorageAdded, StorageStatus::StorageModified, ADDRESS_ALICE, STORAGE_KEY,
-};
 use athena_vm_sdk::call;
 
-// Note: the test harness installs this contract code at ADDRESS_ALICE
-
-// emulate a return value by writing the return value to storage.
-// Athena doesn't support return values yet.
 fn return_value(value: u32) {
-  let key = bytes32_to_32bit_words(STORAGE_KEY);
-  let val: [u32; 8] = [value, 0, 0, 0, 0, 0, 0, 0];
-  let status = write_storage(&key, &val);
```

Comment:
It's nice that this test is now much cleaner, but can we make sure we have another test that tests writing to and reading from storage instead?

Reply:
The tests in https://github.com/athenavm/athena/blob/main/tests/host/src/main.rs cover it already :)